### ISSUE TITLE
dbt-templater doesn't respect trailing newline when exclude_rules = L009

### ISSUE THREAD

**Issue by GClunies** [sqlfluff/sqlfluff#861]
## Expected Behaviour
If `exclude_rules = L009` is included in the `.sqlfluff` config when using `templater = dbt` , then any trailing newlines that are found at the end of a source dbt SQL model should be left as is when running `sqlfluff fix my_dbt_model.sql`. If no trailing newline exists, do nothing.

## Observed Behaviour
When `exclude_rules = L009` is included it the `.sqlfluff` config and using `templater = dbt`, running `sqlfluff fix my_dbt_model.sql` _removes_ any trailing newlines if they exist in the source dbt SQL models.

This is an undesirable outcome for the following reasons:
1. Setting `exclude_rules = L009` in `.sqlfluff` should result in leaving any existing trailing newlines (if they exist) _as is_. Right now, it is _actually enforcing the opposite of L009_ (i.e., no trailing newlines allowed), not _excluding the rule from consideration_ as the config option suggests.
2. Including a trailing newline at the end of files is best practice. Not including these will result in pesky `No newline at end of file` warnings in GitHub. I would _guess_ most users want to keep these trailing newlines

## Steps to Reproduce
Hard to share a _minimal_ example since the dbt templater uses the dbt compiler under the covers, so any example will require a dbt_project.

But the gist of the problem would look like...
A `.sqlfluff` like (per the docs)
```
[sqlfluff]
templater = dbt
# dbt templating does not keep trailing new lines (L009)
exclude_rules = L009
```
A dbt model like `my_dbt_model.sql` (note the trailing newline at end of file)
```sql
select *

from {{ ref('foo_bar_model') }}


```
Running `sqlfluff fix my_dbt_model.sql` will result in (note the trailing newline has been removed!)
```sql
select *

from {{ ref('foo_bar_model') }}
```

## Version
```bash
▶ python --version
Python 3.7.3
```
```bash
▶ sqlfluff --version
sqlfluff, version 0.4.1
```

## Configuration
```
# For SQLFluff Rules reference, see:
# https://docs.sqlfluff.com/en/stable/rules.html#rules-reference
[sqlfluff]
verbose = 0
nocolor = False
dialect = postgres
templater = dbt
rules = None
exclude_rules = L034
recurse = 0
output_line_length = 80
runaway_limit = 10
ignore = parsing
ignore_templated_areas = True

[sqlfluff:indentation]
indented_joins = False
template_blocks_indent = True

# Some rules can be configured directly from the config common to other rules.
[sqlfluff:rules]
tab_space_size = 4
max_line_length = 80
indent_unit = space
comma_style = trailing
allow_scalar = True
single_table_references = consistent
only_aliases = True

# Some rules have their own specific config.
# All SQLFluff rules can be found at: https://docs.sqlfluff.com/en/stable/rules.html#rules-reference
# When a rule in not listed below, we inherit the default behavior from above.
[sqlfluff:rules:L003]
lint_templated_tokens = True

[sqlfluff:rules:L010]  # Keywords
capitalisation_policy = lower

[sqlfluff:rules:L014]  # Unquoted identifiers
capitalisation_policy = lower

[sqlfluff:rules:L016]
# Setting to True allows us to copy/paste long URLs as comments
ignore_comment_lines = True

[sqlfluff:rules:L030]  # Function names
capitalisation_policy = lower

[sqlfluff:rules:L038]
select_clause_trailing_comma = forbid

[sqlfluff:rules:L040]  # Null & Boolean Literals
capitalisation_policy = lower

[sqlfluff:rules:L042]
# By default, allow subqueries in from clauses, but not join clauses.
forbid_subquery_in = join

```
